src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const history = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+      case 'removeProperties':
+        currentState = removeProp(currentState, action);
+        break;
+    }
+    history.push(currentState);
+  }
+
+  function removeProp(obj, { keysToRemove }) {
```

Comment:
The function 'removeProp' could be named more clearly to reflect its purpose, such as 'removePropertiesFromState'.